Binary files amiberry.orig/.git/index and amiberry/.git/index differ
Binary files amiberry.orig/.git/modules/external/capsimg/index and amiberry/.git/modules/external/capsimg/index differ
Binary files amiberry.orig/.git/modules/external/libmpeg2/index and amiberry/.git/modules/external/libmpeg2/index differ
diff -rupN amiberry.orig/Makefile amiberry/Makefile
--- amiberry.orig/Makefile	2022-04-16 16:23:12.481980516 -0400
+++ amiberry/Makefile	2022-04-16 23:36:53.638425501 -0400
@@ -228,30 +228,13 @@ else ifeq ($(PLATFORM),a64)
 else ifeq ($(PLATFORM),x86-64)
 	CPPFLAGS += -DUSE_RENDER_THREAD
 
-# RK3288 e.g. Asus Tinker Board
-# RK3328 e.g. PINE64 Rock64 
-# RK3399 e.g. PINE64 RockPro64 
-# RK3326 e.g. Odroid Go Advance - 32-bit userspace
-else ifneq (,$(findstring RK,$(PLATFORM)))
-	CPPFLAGS += $(CPPFLAGS32) $(NEON_FLAGS)
+# Anbernic RG353P and RG503
+else ifeq ($(PLATFORM),RK3566)
+	CPUFLAGS = -mcpu=cortex-a55
+	CPPFLAGS += $(CPPFLAGS64) -mcpu=cortex-a55 -mtune=cortex-a55 -Ofast -ffast-math -DUSE_RENDER_THREAD
+	AARCH64 = 1
 	HAVE_NEON = 1
 
-	ifneq (,$(findstring RK33,$(PLATFORM)))
-	  CPUFLAGS = -mfloat-abi=hard -mfpu=neon-fp-armv8
-	  ifneq (,$(findstring RK3399,$(PLATFORM)))
-		CPUFLAGS += -mcpu=cortex-a72
-	  else ifneq (,$(findstring RK3328,$(PLATFORM)))
-		CPUFLAGS += -mcpu=cortex-a53
-		CPPFLAGS += -DUSE_RENDER_THREAD
-	  else ifneq (,$(findstring RK3326,$(PLATFORM)))
-		CPUFLAGS += -mcpu=cortex-a35
-		CPPFLAGS += -DUSE_RENDER_THREAD
-	  endif
-	else ifneq (,$(findstring RK3288,$(PLATFORM)))
-	  CPUFLAGS = -mcpu=cortex-a17 -mfloat-abi=hard -mfpu=neon-vfpv4
-	  CPPFLAGS += -DUSE_RENDER_THREAD
-	endif
-
 # sun8i Allwinner H2+ / H3 like Orange PI, Nano PI, Banana PI, Tritium, AlphaCore2, MPCORE-HUB
 else ifeq ($(PLATFORM),sun8i)
 	CPUFLAGS = -mcpu=cortex-a7 -mfpu=neon-vfpv4
